server/src/main/java/org/elasticsearch/indices/recovery/RecoveriesCollection.java

```diff
@@ -275,16 +275,20 @@ protected void doRun() throws Exception {
                 logger.trace("[monitor] no status found for [{}], shutting down", recoveryId);
                 return;
             }
-            long accessTime = status.lastAccessTime();
-            if (accessTime == lastSeenAccessTime) {
-                String message = "no activity after [" + checkInterval + "]";
-                failRecovery(recoveryId,
-                        new RecoveryFailedException(status.state(), message, new ElasticsearchTimeoutException(message)),
-                        true // to be safe, we don't know what go stuck
-                );
-                return;
+            if (status.isRecoveryMonitorEnabled()) {
+                long accessTime = status.lastAccessTime();
+                if (accessTime == lastSeenAccessTime) {
+                    String message = "no activity after [" + checkInterval + "]";
+                    failRecovery(recoveryId,
+                            new RecoveryFailedException(status.state(), message, new ElasticsearchTimeoutException(message)),
+                            true // to be safe, we don't know what go stuck
+                    );
+                    return;
+                }
+                lastSeenAccessTime = accessTime;
+            } else {
+                lastSeenAccessTime = System.nanoTime();
```

Comment:
I think it might be simpler to just fake the progress inside `RecoveryTarget` by returning `System.nanoTime()`?